wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditService.java

```diff
@@ -0,0 +1,194 @@
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.api.audit.server;
+
+import com.codenvy.api.license.CodenvyLicense;
+import com.codenvy.api.license.LicenseException;
+import com.codenvy.api.license.server.CodenvyLicenseManager;
+import com.codenvy.api.permission.server.PermissionManager;
+import com.codenvy.api.permission.server.PermissionsImpl;
+import com.codenvy.api.user.server.dao.AdminUserDao;
+import com.google.common.io.Files;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.compress.utils.IOUtils;
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.rest.Service;
+import org.eclipse.che.api.user.server.model.impl.UserImpl;
+import org.eclipse.che.api.workspace.server.WorkspaceManager;
+import org.eclipse.che.api.workspace.server.model.impl.WorkspaceImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.StreamingOutput;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Defines Audit report REST API.
+ *
+ * @author Igor Vinokur
+ */
+@Path("/audit")
+public class AuditService extends Service {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AuditService.class);
+
+    private final AdminUserDao          adminUserDao;
+    private final WorkspaceManager      workspaceManager;
+    private final PermissionManager     permissionManager;
+    private final CodenvyLicenseManager licenseManager;
+
+    @Inject
+    public AuditService(AdminUserDao adminUserDao,
+                        WorkspaceManager workspaceManager,
+                        PermissionManager permissionManager,
+                        CodenvyLicenseManager licenseManager) {
+        this.adminUserDao = adminUserDao;
+        this.workspaceManager = workspaceManager;
+        this.permissionManager = permissionManager;
+        this.licenseManager = licenseManager;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public StreamingOutput getReport() throws ServerException {
+        File file = initializeFileInTempDirectory();
+
+        printHeader(file);
```

Comment:
I'd move all business logic related to report generation out of `Service` because Service's responsibility is just transport (parsing request, bean validation of parameters, forming response). And report generating is responsibility of class in next layout like `ReportManager`